src/main/java/seedu/tp/studyplan/StudyPlan.java

```diff
@@ -0,0 +1,125 @@
+package seedu.tp.studyplan;
+
+import seedu.tp.exceptions.InvalidFlashcardIndexException;
+import seedu.tp.flashcard.FlashcardList;
+import seedu.tp.ui.Ui;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.FileHandler;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+
+import static seedu.tp.utils.Constants.DATE_FIELD;
+import static seedu.tp.utils.Constants.EMPTY_SPACE;
+import static seedu.tp.utils.Constants.INDEXES_FIELD;
+import static seedu.tp.utils.Constants.LOG_FOLDER;
+
+public class StudyPlan {
+
+    protected static final Logger LOGGER = Logger.getLogger(StudyPlan.class.getName());
+    private static final String FILE_PATH = LOG_FOLDER + "study_plan.log";
+
+    private Map<LocalDate, List<Integer>> studyPlan;
+
+    /**
+     * Constructor for StudyPlan.
+     */
+    public StudyPlan() {
+        this.studyPlan = new TreeMap<>();
+    }
+
```

Comment:
I think it may be clearer if StudyPlan was named StudyPlanList instead. StudyPlan makes it sound as though the class is only in charge of one study plan, when actually studyPlan is a map containing all the study plans for all dates.

Reply:
Makes sense! I'll update the naming :)